### PR TITLE
feat(vm): add failure message to condition if migration is failed

### PR DESCRIPTION
**Pull request by yaroslavborbat**
## Description
add failure message to condition if migration is failed

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [X] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/virtualization-artifact/pkg/controller/vm/internal/util.go [hunk 1]
```diff
@@ -159,3 +159,9 @@ func isInternalVirtualMachineError(phase virtv1.VirtualMachinePrintableStatus) b
 func podFinal(pod corev1.Pod) bool {
 	return pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed
 }
+
+func newSetCondition(vm *virtv2.VirtualMachine) func(c conditions.Conder) {
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/vm/internal/util.go:163]: Don't see the reason to wrap here
Just use `conditions.SetCondition(c, &vm.Status.Conditions)` and keep it simple

File: images/virtualization-artifact/pkg/controller/vm/internal/lifecycle.go [hunk 2]
```diff
@@ -127,83 +127,80 @@ func (h *LifeCycleHandler) syncMigrationState(vm *virtv2.VirtualMachine, kvvm *v
...
+
+	case kvvmi != nil && kvvmi.Status.MigrationState != nil &&
+		kvvmi.Status.MigrationState.EndTimestamp != nil &&
+		kvvmi.Status.MigrationState.Failed:
+
+		msg := kvvmi.Status.MigrationState.FailureReason
+		cbMigrating.Status(metav1.ConditionFalse).
+			Reason(vmcondition.ReasonLastMigrationFinishedWithError).
+			Message(msg)
+		setCondition(cbMigrating)
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/vm/internal/lifecycle.go:150]: You don't do return, is it expected? 
Looks like this error can be overridden below

**Review by Isteb4k — CHANGES_REQUESTED** [review]

File: images/virtualization-artifact/pkg/controller/vm/internal/util.go (hunk 1 above)
Comment (yaroslavborbat) [reply images/virtualization-artifact/pkg/controller/vm/internal/util.go:163]: ok

File: images/virtualization-artifact/pkg/controller/vm/internal/lifecycle.go (hunk 2 above)
Comment (yaroslavborbat) [reply images/virtualization-artifact/pkg/controller/vm/internal/lifecycle.go:150]: this is expected. the syncMigrationState method updates 2 conditions, so I don't return

**Review by Isteb4k — APPROVED** [review]

**Pull request merged**